sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -227,14 +257,14 @@ public Aggregation toDruidAggregation(
           aggregatorType.name(),
           null,
           SqlKind.OTHER_FUNCTION,
-          ReturnTypes.ARG0,
+          new EarliestLatestReturnTypeInference(0),
           InferTypes.RETURN_TYPE,
           OperandTypes.or(
               OperandTypes.NUMERIC,
               OperandTypes.BOOLEAN,
               OperandTypes.sequence(
                   "'" + aggregatorType.name() + "(expr, maxBytesPerString)'\n",
-                  OperandTypes.STRING,
+                  OperandTypes.ANY,
```

Comment:
shouldn't this remain `STRING` ?

Reply:
When we specify "stringFirst/stringLast" in ingestion MetricsSpec, the type for the columns stored in segments is `OTHER`. With `STRING` type here, the SQL will report an error when trying to call `earliest/latest` on those columns.

Reply:
If you have any suggestions on where is a good place to add a test for this, that would be great!

Reply:
Usually I sql add tests to `CalciteQueryTest` You can see `testEarliestAggregators` as an example. You could write a test in here to validate the SQL translation works, but I think you need to update `CalciteTests#INDEX_SCHEMA_DIFFERENT_DIM3_M1_TYPES` to add an earliest / latest aggregator. Then you should be able to test a SQL test in `CalciteQueryTest` - hope this helps